vendor/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -727,3 +728,11 @@ type DeletedFinalStateUnknown struct {
 	Key string
 	Obj interface{}
 }
+
+// DeletedStronglyTypedFinalStateUnknown is like DeletedFinalStateUnknown, but
+// uses a types.NamespacedName as the key. This is useful for objects that are
+// namespaced, as it allows for more precise identification of the object.
+type DeletedStronglyTypedFinalStateUnknown struct {
```

Comment:
Ditto re: vendored dependencies.
We can't really replace `DeletedFinalStateUnknown` with a proprietary type as cache implementation depends on it.